### DISCUSSION TITLE
Filter fetched data via useSWR

### DISCUSSION THREAD

**Discussion by alex96dev** [neuefische/questions#822] [Web] [answered]
I'm fetching users from the mongoDB via useSWR and get the result: 

![Bildschirmfoto 2024-03-27 um 16 27 12](https://github.com/neuefische/questions/assets/138215839/8152923a-730c-4e3b-9dbd-c54b9dc40ef5)

I want to filter the friends array within the response-array. In order to do that I used the .find() method within my useEffect like this:

useEffect(() => {
    async function fetchData() {
      const url = `https://api.weatherapi.com/v1/current.json?key=27d7a8c529bf4b43af9113412242203&q=${city}&aqi=no`;

      try {
        const response = await fetch(url);
        const result = await response.json();

        const mainTemp = result.current.temp_c;
        const condition = result.current.condition.icon;

        setWeather(mainTemp);
        setCondition(condition);
        if (appUsers) {
          const user = appUsers.find((user) => user._id === session?.user?.id);
          console.log("user:", user);

          if (user && user.friends) {
            console.log("user.friends:", user.friends);
          } else {
            console.log("User or user.friends is undefined");
          }
        }
      } catch (error) {
        console.error(error);
      }
    }
    if (appUsers) {
      fetchData();
    }
  }, [appUsers]);

The logic itself works but only with a confusing behavior. When logging in it seems that appUsers is undefined and no console.log of the if statement shows up. Besides I get 4 or more console.logs as shown in the screenshot above and I don't know why. The confusing part is that if I navigate through pages and go back to the homepage via buttons I can see the console.logs and I get the console.logs with the user.friends data...
Then it looks like this: 

<img width="1355" alt="Bildschirmfoto 2024-03-28 um 10 03 27" src="https://github.com/neuefische/questions/assets/138215839/56c4f48c-4151-4dc9-a953-1909f34d2063">

**Comment by NikRoe** [comment]
Hi Alex,

can you send me a PR to the current status? So that I can see more of the code.

Thanks in advance.

(In general, having console.logs being shown several times is now, since we work with the strict mode that executes everything twice. Also having console.logs in your useEffect will add do this, due to the appUsers dependency)

**Reply by alex96dev** [reply]
Sure. Here is the PR: 
https://github.com/alex96dev/Easy-Meetup-with-Friends/pull/78

**Comment by Kheirah** [comment, marked as answer]
Yes, @NikRoe is right about the strict mode while in development. Your code is **always** running _twice._ Once on the server and once on the client. 

In production, your code only runs once, as you'd expect. There are only 2 logs instead of 4. Once for the first render and once for the second render after running useEffect in the first render.

<img width="518" alt="image" src="https://github.com/neuefische/questions/assets/22839911/1ea00058-32a8-4e63-8940-9ea79c087fd2">

Your app works as expected I'd reckon.😊

**Comment by alex96dev** [comment]
Hey guys, 
first of all thanks for the response!

I currently get way more than 4 responses of the same type. I think this occurs due to the logic regarding the states and their updating logic. It is kind of annoying and unnecessary but the logic works anyways..

I just resolved the problem regarding the confusing behavior in the way that I fixed a typo in the code. Their was an id that I had to address with "_id" instead of "id". One of the classic ones i guess...

Thanks for the help. I will continue my work now.
